### PR TITLE
APPS-1241 Add concurrent scans limiter

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

File: io/aerospike/record_reader_test.go [hunk 1]
```diff
@@ -70,6 +71,7 @@ func (suite *readersTestSuite) TestAerospikeRecordReader() {
 	)
 
 	reader := NewRecordReader(
+		context.TODO(),
```
Comment (filkeith) [inline io/aerospike/record_reader_test.go:74]: If it is test why not to use context.Background(). As context.TODO() is used as TODO for future, when we don't know what context to use right now, and plan to replace it in future. But it is test, we don't have such problem here

File: client.go [hunk 2]
```diff
@@ -90,33 +91,60 @@ type AerospikeClient interface {
 type Client struct {
 	aerospikeClient AerospikeClient
 	logger          *slog.Logger
+	scanLimiter     *semaphore.Weighted
 	id              string
 }
 
+type Option func(*Client)
```
Comment (reugn) [inline client.go:98]:
```suggestion
// Option is a functional option that allows configuring the [Client].
type Option func(*Client)
```

File: client.go [hunk 3]
```diff
@@ -90,33 +91,60 @@ type AerospikeClient interface {
...
+
+// WithID sets the ID for the Client.
```
Comment (reugn) [inline client.go:100]:
```suggestion
// WithID sets the ID for the [Client].
```

File: client.go [hunk 4]
```diff
@@ -90,33 +91,60 @@ type AerospikeClient interface {
...
+func WithID(id string) Option {
+	return func(c *Client) {
+		c.id = id
+	}
+}
+
+// WithLogger sets the logger for the Client.
```
Comment (reugn) [inline client.go:107]:
```suggestion
// WithLogger sets the logger for the [Client].
```

File: client.go [hunk 5]
```diff
@@ -90,33 +91,60 @@ type AerospikeClient interface {
...
+func WithLogger(logger *slog.Logger) Option {
+	return func(c *Client) {
+		c.logger = logger
+	}
+}
+
+// WithScanLimiter sets the scanLimiter for the Client.
```
Comment (reugn) [inline client.go:114]:
```suggestion
// WithScanLimiter sets the scan limiter for the [Client].
```

File: client.go [hunk 6]
```diff
@@ -90,33 +91,60 @@ type AerospikeClient interface {
...
+	return func(c *Client) {
+		c.scanLimiter = sem
+	}
+}
+
 // NewClient creates a new backup client.
+// Options:
 //   - ac is the aerospike client to use for backup and restore operations.
 //   - id is an identifier for the client.
 //   - logger is the logger that this client will log to.
```
Comment (reugn) [inline client.go:125]:
```suggestion
// NewClient creates a new backup client.
//   - ac is the aerospike client to use for backup and restore operations.
//
// Options:
//   - [WithID] to set an identifier for the client.
//   - [WithLogger] to set a logger that this client will log to.
//   - [WithScanLimiter] to set a semaphore that is used to limit number of
//     concurrent scans.
```

File: client.go (hunk 2 above)
Comment (reugn) [inline client.go:98]: It's better to name it `ClientOption` or `ClientOpt`.

File: examples/aws/s3/main.go [hunk 7]
```diff
@@ -17,12 +17,10 @@ package main
 import (
 	"context"
 	"fmt"
-	"log"
-	"log/slog"
-
 	"github.com/aerospike/aerospike-client-go/v7"
 	"github.com/aerospike/backup-go"
 	"github.com/aerospike/backup-go/io/aws/s3"
+	"log"
```
Comment (reugn) [inline examples/aws/s3/main.go:23]: Format imports.

File: examples/readme/main.go [hunk 8]
```diff
@@ -16,11 +16,9 @@ package main
 
 import (
 	"context"
-	"log"
-	"log/slog"
-
 	"github.com/aerospike/aerospike-client-go/v7"
 	"github.com/aerospike/backup-go"
+	"log"
```
Comment (reugn) [inline examples/readme/main.go:21]: Format imports.

**Review by reugn — COMMENTED** [review]
* Add coverage for the client options.

**Review by reugn — APPROVED** [review]

**Pull request merged**